src/hotspot/cpu/aarch64/aarch64_ad.m4

```diff
@@ -539,4 +542,82 @@ dnl
 ADD_SUB_ZERO_EXTEND_SHIFT(I,255,Sub,subw,uxtb)
 ADD_SUB_ZERO_EXTEND_SHIFT(I,65535,Sub,subw,uxth)
 dnl
+define(`CMOV_INSN', `// This pattern is automatically generated from aarch64_ad.m4
+// DO NOT EDIT ANYTHING IN THIS SECTION OF THE FILE
+instruct cmov$1_reg_reg_$3(iReg$1NoSp dst, iReg$1 src1, iReg$1 src2, rFlagsReg cr)
+%{
+  effect(DEF dst, USE src1, USE src2, USE cr);
+
+  ins_cost(INSN_COST * 2);
+  format %{ "$2 $dst, $src1, $src2 $3\t"  %}
+
+  ins_encode %{
+    __ $2(as_Register($dst$$reg),
```

Comment:
I wonder if it would be better to use `$dst$$Register` here and several other sites in this patch.
```suggestion
    __ $2($dst$$Register,
```

Reply:
Changed as suggested. I also made some cleanup in .m4 to and fixed minor .ad-.m4 mismatches.